agent/structs/config_entry_routes.go

```diff
@@ -0,0 +1,127 @@
+package structs
+
+import "github.com/hashicorp/consul/acl"
+
+// HTTPRouteConfigEntry stub
+type HTTPRouteConfigEntry struct {
+	Kind string
+	Name string
+
+	Meta               map[string]string `json:",omitempty"`
+	acl.EnterpriseMeta `hcl:",squash" mapstructure:",squash"`
+	RaftIndex
+}
+
+func (e *HTTPRouteConfigEntry) GetKind() string {
+	return HTTPRoute
+}
+
+func (e *HTTPRouteConfigEntry) GetName() string {
+	if e == nil {
+		return ""
+	}
+	return e.Name
+}
+
+func (e *HTTPRouteConfigEntry) Normalize() error {
+	return nil
+}
+
+func (e *HTTPRouteConfigEntry) Validate() error {
+	return nil
+}
+
+func (e *HTTPRouteConfigEntry) CanRead(authz acl.Authorizer) error {
+	var authzContext acl.AuthorizerContext
+	e.FillAuthzContext(&authzContext)
+	return authz.ToAllowAuthorizer().MeshReadAllowed(&authzContext)
+}
+
+func (e *HTTPRouteConfigEntry) CanWrite(authz acl.Authorizer) error {
+	var authzContext acl.AuthorizerContext
+	e.FillAuthzContext(&authzContext)
+	return authz.ToAllowAuthorizer().MeshWriteAllowed(&authzContext)
+}
+
+func (e *HTTPRouteConfigEntry) GetMeta() map[string]string {
+	if e == nil {
+		return nil
+	}
+	return e.Meta
+}
+
+func (e *HTTPRouteConfigEntry) GetEnterpriseMeta() *acl.EnterpriseMeta {
+	if e == nil {
+		return nil
+	}
+	return &e.EnterpriseMeta
+}
+
+func (e *HTTPRouteConfigEntry) GetRaftIndex() *RaftIndex {
+	if e == nil {
+		return &RaftIndex{}
+	}
+	return &e.RaftIndex
+}
+
+// TCPRouteConfigEntry stub
+type TCPRouteConfigEntry struct {
+	Kind string
+	Name string
+
+	Meta               map[string]string `json:",omitempty"`
+	acl.EnterpriseMeta `hcl:",squash" mapstructure:",squash"`
+	RaftIndex
```

Comment:
nit: missing all docstrings here, do we normally include them on the server-side structs or just on the client side API structs (since I'm assuming that's what most people are actually looking at)

Reply:
This is super hit or miss on existing config entry types. Some include no docstrings at all, others include docs for `Kind` and `Name` only, and none seem to include docstrings for `Meta`, `EnterpriseMeta` or `RaftIndex`.

Reply:
😞 ok, that's fine then, we can keep the server-side stuff as-is.